examples/running_mean/main.rs

```diff
@@ -0,0 +1,59 @@
+extern crate linfa;
+extern crate ndarray;
+extern crate ndarray_rand;
+extern crate rand;
+#[macro_use]
+extern crate derive_more;
+
+use crate::standard_scaler::{Config, OnlineOptimizer, ScalingError, StandardScaler};
+use linfa::{Fit, IncrementalFit, Transformer};
+use ndarray::{stack, Array1, ArrayBase, Axis, Data, Ix1};
+use ndarray_rand::RandomExt;
+use rand::distributions::Uniform;
+
+mod standard_scaler;
+
+fn generate_batch(n_samples: usize) -> (Array1<f64>, Array1<f64>) {
+    let distribution = Uniform::new(0., 10.);
+    let x = Array1::random(n_samples, distribution);
+    let y = Array1::random(n_samples, distribution);
+    (x, y)
+}
+
+fn check<S>(scaler: &StandardScaler, x: &ArrayBase<S, Ix1>) -> Result<(), ScalingError>
+where
+    S: Data<Elem = f64>,
+{
+    let old_batch_mean = x.mean_axis(Axis(0)).into_scalar();
+    let new_batch_mean = scaler.transform(&x)?.mean_axis(Axis(0)).into_scalar();
+    let old_batch_std = x.std_axis(Axis(0), 1.).into_scalar();
+    let new_batch_std = scaler.transform(&x)?.std_axis(Axis(0), 1.).into_scalar();
+    println!(
+        "The mean.\nBefore scaling: {:?}\nAfter scaling: {:?}\n",
+        old_batch_mean, new_batch_mean
+    );
+    println!(
+        "The std deviation.\nBefore scaling: {:?}\nAfter scaling: {:?}\n",
+        old_batch_std, new_batch_std
+    );
+    Ok(())
+}
+
+/// Run it with: cargo run --example running_mean
+fn main() -> Result<(), ScalingError> {
+    let n_samples = 20;
+    let (x, y) = generate_batch(n_samples);
+
+    let mut optimizer = OnlineOptimizer::default();
+    let standard_scaler = optimizer.fit(&x, &y, Config::default())?;
```

Comment:
Passing the config in fit time might make difficult to compose estimators. Say if the esimator is a pipeline of estimators we wouldn't want to pass all the config in a fit. Having two steps a) building the pipeline b) fitting it is more natural IMO

Reply:
The two things are not mutually exclusive I'd say. You could compose the configuration of all steps in the pipeline and then pass that in when you want to fit it, it shouldn't look very different.
But I have yet to actually prototype it, so take it with a grain of salt.